### PR TITLE
Нечесанов - Лабораторная работа #1

### PR DESCRIPTION
**Pull request by VladimirNechesanov**

### CODE REVIEW

**Comment by VladimirNechesanov** [comment]
@katerina394
@xnd-r

File: code/sorting/sorting.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    def insertion_sort(self):
+        nums = self.convert_to_array_of_int()[:]
+
+        for index in range(1, len(nums)):
+            curr_val = nums[index]
+            curr_pos = index
+            while curr_pos > 0 and nums[curr_pos - 1] > curr_val:
+                nums[curr_pos] = nums[curr_pos - 1]
+                curr_pos = curr_pos - 1
```
Comment (xnd-r) [inline code/sorting/sorting.py:29]: could be good `curr_pos -= 1`

File: code/sorting/sorting.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+stringType = type('')
```
Comment (xnd-r) [inline code/sorting/sorting.py:1]: Camel case and snake case mixing

File: code/sorting/sorting.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    def convert_to_array_of_int(self):
+        array_of_int = []
+
+        for i in self.array:
+            if type(i) == stringType and not i.isalpha():
+                array_of_int.append(int(i))
+            if type(i) == stringType and not i.isalpha():
+                continue
+            if type(i) != stringType:
+                array_of_int.append(i)
```
Comment (katerina394) [inline code/sorting/sorting.py:17]: unreasonable duplication of "if" branches and wrong logic (.isalpha() return false for string "123abc")

```suggestion
            if isinstance(i, str) and i.isnumeric():
                array_of_int.append(int(i))
```

**Comment by katerina394** [comment]
Also, add a check that parameter array of __init__ has type "list"

File: code/sorting/test_sorting.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import unittest
+
+from sorting.sorting import Sorting
+
+FIRST_CASE = [0, 7, 4]
+SECOND_CASE = [0, 45, 1,  8, 2, 4]
+THIRD_CASE = [0, 4, '45', '1', 2, 8]
+FOURTH_CASE = [1, '45', 3, 5, 'wef']
+
```
Comment (xnd-r) [inline code/sorting/test_sorting.py:9]: Can you please add tests with float/negative values

File: code/sorting/sorting.py (hunk 1 above)
Comment (VladimirNechesanov) [reply code/sorting/sorting.py]: done

File: code/sorting/sorting.py (hunk 2 above)
Comment (VladimirNechesanov) [reply code/sorting/sorting.py]: done

File: code/sorting/sorting.py (hunk 3 above)
Comment (VladimirNechesanov) [reply code/sorting/sorting.py]: done

File: code/sorting/test_sorting.py (hunk 4 above)
Comment (VladimirNechesanov) [reply code/sorting/test_sorting.py:12]: done

**Comment by xnd-r** [comment]
Oke! 👍

**Comment by katerina394** [comment]
👍

**Comment by VladimirNechesanov** [comment]
@kirill-korniakov please review

**Review by sawirani — APPROVED** [review]
